dev/fattest.simplicity/src/componenttest/rules/repeater/JakartaEEAction.java

```diff
@@ -276,13 +276,7 @@ public static void transformApp(Path appPath, Path newAppPath, EEVersion eeVersi
      * @param transformationRulesAppend The map with the additional transformation rules to add
      */
     public static void transformApp(Path appPath, Path newAppPath, Map<String, String> transformationRulesAppend) {
-        if (RepeatTestFilter.isRepeatActionActive(EE9_ACTION_ID)) {
-            JakartaEE9Action.transformApplication(appPath, newAppPath, transformationRulesAppend);
-        } else if (RepeatTestFilter.isRepeatActionActive(EE10_ACTION_ID)) {
-            JakartaEE10Action.transformApplication(appPath, newAppPath, transformationRulesAppend);
-        } else if (RepeatTestFilter.isRepeatActionActive(EE11_ACTION_ID)) {
-            JakartaEE11Action.transformApplication(appPath, newAppPath, transformationRulesAppend);
-        }
+        RepeatTestFilter.getMostRecentRepeatAction().transformApplication(appPath, newAppPath, transformationRulesAppend);
```

Comment:
This logic has been changed ... it now only checks only the top action on the stack rather than all of them. This change may break some tests.

Reply:
Yeah I don't like this at all.  I don't understand why this change is necessary.

Reply:
Actually the logic hasn't changed. Due to the bizarre code for repeat tests the stack only ever has one repeat test action on it at any given time. So its already only checking the top action.

As for why its necessary. Currently if you do `FeatureReplacementAction.EE10_FEATURES().withID("my_id")` apps will not be transformed because transformation runs based on string comparisons from the IDs. This is very unintuitive behaviour and this PR was created in response to someone tripping over it. With this new code transformation is a property of the class itself which leads to far more logical behaviour.